app/src/main/java/com/asfoundation/wallet/promo_code/bottom_sheet/entry/PromoCodeBottomSheetFragment.kt

```diff
@@ -96,10 +96,10 @@ class PromoCodeBottomSheetFragment : BottomSheetDialogFragment(),
         }
       }
       is Async.Fail -> {
-        handleErrorState(FailedPromoCode.InvalidCode(clickAsync.error.throwable))
+        handleErrorState(ValidityState.NOT_ADDED)
```

Comment:
Please don't do this.
Do not leak the business entity to the view and use `PromoCodeResult`.
And same for everywhere in this file

Reply:
As I said in the other comment, this PromoCodeResult is used for deciding what to show when opening the screen with the already stored promo code, so both entities were serving different purposes.
What I could do is to map the ValidityState to a new model for the view. Is that ok to you?